sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultResourceSpec.java

```diff
@@ -23,6 +24,7 @@
     @Size(min = 1)
     private final String name;
     @NotNull
+    @NonZeroScalarProtoValue
```

Comment:
I'd change this to `PositiveScalar` or something similar.  `NonZero` could mean negative, and the annotation as a whole is a bit verbose.